lib/galaxy/config/sample/container_resolvers.yml.sample

```diff
@@ -27,69 +27,62 @@
 # if ``enable_mulled_containers`` is set in ``galaxy.yml`` (which is the default).
 
 # get a container description to a cached mulled docker container
-# - checks in the docker cache if the image is there
 - type: cached_mulled
+  # This resolver type will check the image cache for `quay.io/NAMESPACE/containername`
+  # The image cache can be populated by `mulled_*` resolvers or
+  # with a manual `docker pull quay.io/NAMESPACE/containername`.
   namespace: biocontainers
 - type: cached_mulled
   namespace: local
-  # This container resolver has two arguments
-  # the resolver will check the image cache for `quay.io/NAMESPACE/containername`
-  # so if one (or another container resolver) did a
-  # `docker pull quay.io/NAMESPACE/containername` then the image should be found
-  # Note the local namespace is currently "abused" to refer to locally built
-  # images (since there is no quay.io/local/)
+  # The local namespace refers to locally built images that are prefixed with quay.io/local/.
   #namespace: biocontainers
   # the version of the mulled hashing function (v2/v1)
   #hash_func: v2
 
 
 # get a container description for a cached mulled singularity container
-# - checks in the local cache directory if the image file exists and returns the path if so
+# checks if the image file exists in `cache_directory`
 - type: cached_mulled_singularity
   namespace: biocontainers
-- type: cached_mulled_singularity
-  namespace: local
-  # NOTE that the default container config specifies the namespace, but it is
-  # currently not used in the code, i.e. the two resolvers are redundant
   #
   #cache_directory: database/container_cache/singularity/mulled
   #
   # the method for caching directory listings (not the method for image caching)
-  # can be uncached, dir_mtime (the latter only determines the directory listing
-  # if the mtime of the directory changed)
-  # cache_directory_cacher_type: uncached
-  #
-  # the version of the mulled hashing function (v2/v1)
-  #hash_func: v2
+  # can be uncached or dir_mtime (the latter only determines the directory listing
+  # if the modification time of the directory changed)
+  #cache_directory_cacher_type: uncached
+
+- type: cached_mulled_singularity
+  namespace: local
+  #cache_directory_cacher_type: uncached
 
-# Build a mulled container description to quay.io/NAMESPACE/MULLED_HASH where the
+# Resolves container images from quay.io/NAMESPACE/MULLED_HASH where the
 # mulled hash describes which packages and versions should be in the container
 #
-# if the corresponding CLI (docker) is available also the image
-# will be pulled, i.e. the image will be in dockers image cache
+# If the docker CLI is available the image will be pulled.
+#
+# If auto_install is set to true the container resolver pull the image and point
```

Comment:
Unfortunately this does not describe the current state. The image is pulled irrespective of `auto_install`. The only difference that `auto_install` makes is if the URI or the cached image is returned.

xref https://github.com/galaxyproject/galaxy/issues/15717

Reply:
... because the image gets pulled in the handler (auto_install=True).

Reply:
but feel free to suggest an improvement to `depending on the value of auto_install the container resolver will point`

Reply:
Then my suggestion is to remove: 

```
If auto_install is set to true the container resolver pull the image and point
to the cached image, otherwise the reference to quay.io/NAMESPACE/MULLED_HASH is passed
on to the job script, where the image can be pulled.
```

> ... because the image gets pulled in the handler (auto_install=True). 

This happens completely independent of the value of `auto_install` (probably a bug). 

In the end `auto_install` makes absolutely no difference for `mulled`
since the name of the cached image is equal to the image name
ie `quay.io/NAMESPACE/MULLED_HASH` and also the image is pulled
anyway (if the resolver is called during job preparation of container installation).
ADDITIONALLY the container is pulled during execution on the compute node
if its absent. 

The only difference for `mulled_singularity` where the name differs (but the
image is also pulled during job preparation and and container installation).

Reply:
Ugh, you're right, it looks like only the build_* resolvers use self.auto_install correctly ... . Alright let's document it the way it is.